org.eclipse.dawnsci.hdf5/src/org/eclipse/dawnsci/hdf5/nexus/NexusFileHDF5.java

```diff
@@ -334,6 +339,36 @@ public GroupNode getGroup(String augmentedPath, boolean createPathIfNecessary) t
 		return (GroupNode) (node.name == null ? null : (node.node));
 	}
 
+	@Override
+	public Node getNode(String augmentedPath) throws NexusException {
+		String plainPath = NexusUtils.stripAugmentedPath(augmentedPath);
+		if (!isPathValid(plainPath)) {
+			return null;
+		}
+		NodeLink link = tree.findNodeLink(plainPath);
+		if (link != null) {
+			if (link.isDestinationData()) {
+				return getData(augmentedPath);
+			} else if (link.isDestinationGroup()) {
+				return getGroup(augmentedPath, true);
+			}
+		} else {
+			// test if root
+			NodeData node = getNode(plainPath, false);
+			NodeType type = node.type;
+			if (type == NodeType.DATASET) {
+				String parentPath = plainPath.substring(0, plainPath.lastIndexOf(Node.SEPARATOR));
+				String name = plainPath.substring(plainPath.lastIndexOf(Node.SEPARATOR) + 1, plainPath.length());
+				GroupNode parentNode = getGroup(parentPath, true);
+				return getData(parentNode, name);
+			} else if (type == NodeType.GROUP) {
+				GroupNode groupNode = getGroup(plainPath, true);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "groupNode". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)